styx-scheduler-service/src/main/java/com/spotify/styx/state/StateUtil.java

```diff
@@ -120,11 +120,10 @@ public static boolean hasTimedOut(Optional<Workflow> workflowOpt, RunState runSt
                                           .flatMap(workflow -> workflow.configuration().runningTimeout())
                                           .orElse(timeout)
                                       : timeout;
-    final Duration sanitizedTimeout = effectiveTimeout.compareTo(timeout) < 0 ? effectiveTimeout : timeout;
```

Comment:
We could keep this I guess and use the new running_max value

Reply:
I think we should keep this.

Reply:
If it is too troublesome, I think it is fine we skip this.

Reply:
I added it back